src/processor/reconnection-graph.service.ts

```diff
@@ -46,63 +61,61 @@ export class ReconnectionGraphService {
+      exportedUpdates = exportedUpdates.filter((update) => update.ownerDsnpUserId === dsnpUserStr);
+
       // loop over exportUpdates and collect Updates vs userId
       exportedUpdates.forEach((bundle) => {
-        const ownerMsaId: MessageSourceId = this.api.registry.createType('MessageSourceId', bundle.ownerDsnpUserId);
-        if (mapUserIdToUpdates.has(ownerMsaId.toString())) {
-          const updates = mapUserIdToUpdates.get(ownerMsaId.toString()) || [];
+        const ownerMsaId: MessageSourceId = this.blockchainService.api.registry.createType('MessageSourceId', bundle.ownerDsnpUserId);
+        if (mapUserIdToUpdates.has(ownerMsaId)) {
+          const updates = mapUserIdToUpdates.get(ownerMsaId) || [];
           updates.push(bundle);
-          mapUserIdToUpdates.set(ownerMsaId.toString(), updates);
+          mapUserIdToUpdates.set(ownerMsaId, updates);
         } else {
-          mapUserIdToUpdates.set(ownerMsaId.toString(), [bundle]);
+          mapUserIdToUpdates.set(ownerMsaId, [bundle]);
         }
       });
 
-      for (const [userId, updates] of mapUserIdToUpdates.entries()) {
+      for (const [ownerMsaId, updates] of mapUserIdToUpdates.entries()) {
         let batch: SubmittableExtrinsic<'rxjs', ISubmittableResult>[] = [];
         let batchCount = 0;
-        const promises: Promise<any>[] = [];
+        const promises: Promise<ParsedEventResult>[] = [];
         updates.forEach((bundle) => {
-          const ownerMsaId: MessageSourceId = this.api.registry.createType('MessageSourceId', userId);
           switch (bundle.type) {
             case 'PersistPage':
-              const payload: any = Array.from(Array.prototype.slice.call(bundle.payload));
-              const upsertPageCall = ExtrinsicHelper.api.tx.statefulStorage.upsertPage(ownerMsaId, bundle.schemaId, bundle.pageId, bundle.prevHash, payload);
-
-              batch.push(upsertPageCall);
+              batch.push(
+                this.blockchainService.createExtrinsicCall(
+                  { pallet: 'statefulStorage', extrinsic: 'upsertPage' },
+                  ownerMsaId,
+                  bundle.schemaId,
+                  bundle.pageId,
+                  bundle.prevHash,
+                  Array.from(Array.prototype.slice.call(bundle.payload)),
+                ),
+              );
               batchCount++;
 
               // If the batch size exceeds the capacityBatchLimit, send the batch to the chain
               if (batchCount === this.capacityBatchLimit) {
-                const payWithCapacityBatchAllOp = ExtrinsicHelper.payWithCapacityBatchAll(providerKeys, batch);
-                promises.push(payWithCapacityBatchAllOp.signAndSend());
                 // Reset the batch and count for the next batch
+                promises.push(
+                  this.blockchainService.createExtrinsic({ pallet: 'frequencyTxPayment', extrinsic: 'payWithCapacityBatchAll' }, {}, providerKeys, batch).signAndSend(),
```

Comment:
If later on you're going to assign `const [event, eventMap] = /* result of this promise */`, then you'll want to pass an event type as the second argument, otherwise `event` will always come back undefined. Like so:
```
{ eventPallet: 'utility', event: 'BatchCompleted' }
```